src/parser.js

```diff
@@ -0,0 +1,520 @@
+const _ = require('underscore')
+const marked = require('marked')
+const Doc = require('./doc')
+const {getLinkMatch, multiplyString} = require('./utils')
+
+const SpecialHeadingDepth = 2
+const SpecialHeadings = /^Arguments|Events|Examples/
+
+const VisibilityRegexStr = '^\\s*([a-zA-Z]+):\\s*'
+const VisibilityRegex = new RegExp(VisibilityRegexStr)
+const ReturnsRegex = new RegExp(`(${VisibilityRegexStr})?\\s*Returns`)
+const ArgumentListItemRegex = /^\s*`([\w\\.-]+)`(\s*[:-])?(\s*\(optional\))?\s*/
+
+/*
+Section: Parsing
+
+Translating things from markdown into our json format.
+*/
+
+// Public: Parses a docString
+//
+// * `docString` a string from the documented object to be parsed
+//
+// Returns a {Doc} object
+const parse = function (docString) {
+  const lexer = new marked.Lexer()
+  const tokens = lexer.lex(docString)
+  const firstToken = _.first(tokens)
```

Comment:
`tokens[0]`?

Reply:
That's what I would use everywhere myself, the code was previously using `_.first` everywhere though. I'll happily remove it 😉.